### PR TITLE
Add Data.Product.Properties.map-cong₂

### PR DESCRIPTION
**Pull request by Taneb**

### CODE REVIEW

File: src/Data/Product/Properties.agda [hunk 1]
```diff
@@ -53,6 +55,9 @@ module _ {B : A → Set b} where
 ,-injective : ∀ {a c : A} {b d : B} → (a , b) ≡ (c , d) → a ≡ c × b ≡ d
 ,-injective refl = refl , refl
 
+map-cong₂ : ∀ {f g : A → C} {h i : B → D} → f ≗ g → h ≗ i → map f h ≗ map g i
+map-cong₂ p q (x , y) = cong₂ _,_ (p x) (q y)
```
Comment (MatthewDaggitt) [inline src/Data/Product/Properties.agda:59]: Very minor bikeshedding but could we rename `p` and `q` to `f≗g` and `h≗i` instead?

File: src/Data/Product/Properties.agda [hunk 2]
```diff
@@ -53,6 +55,9 @@ module _ {B : A → Set b} where
 ,-injective : ∀ {a c : A} {b d : B} → (a , b) ≡ (c , d) → a ≡ c × b ≡ d
 ,-injective refl = refl , refl
 
+map-cong₂ : ∀ {f g : A → C} {h i : B → D} → f ≗ g → h ≗ i → map f h ≗ map g i
```
Comment (MatthewDaggitt) [inline src/Data/Product/Properties.agda:58]: Hmm I think this should be called just `map-cong`? And then the subscripts can be reserved for mapping over either the first or second projections?

Comment (Taneb) [reply src/Data/Product/Properties.agda:58]: I think you might be right. Data.Sum.Properties uses that naming convention.

File: src/Data/Product/Properties.agda (hunk 1 above)
Comment (Taneb) [reply src/Data/Product/Properties.agda:59]: Sure

**Pull request merged**